src/RA_Interface.cpp

```diff
@@ -366,15 +367,13 @@ void WriteBufferToFile(const char* sFile, const char* sBuffer, int nBytes)
 void FetchIntegrationFromWeb()
 {
     const int MAX_SIZE = 2 * 1024 * 1024;
-    char* buffer = new char[MAX_SIZE];
-    if (buffer != nullptr)
+
+    if (auto buffer{ std::make_unique<char[]>(MAX_SIZE) }; buffer != nullptr)
     {
```

Comment:
Save an extra step by destroying it earlier.
It really should be intuitive.
`if precondition; is condition satisfied?` then `postcondition` is automatically handled (destroying the owned pointer in the if..else if..else whatever it may be chain).